src/main/java/se/sundsvall/contract/integration/db/AttachmentRepository.java

```diff
@@ -0,0 +1,19 @@
+package se.sundsvall.contract.integration.db;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import se.sundsvall.contract.integration.db.model.AttachmentEntity;
+
+import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
+
+@CircuitBreaker(name = "attachmentRepository")
+public interface AttachmentRepository extends JpaRepository<AttachmentEntity, Long> {
+
+    List<AttachmentEntity> findAllByContractId(String contractId);
+    void deleteAllByContractId(String contractId);
```

Comment:
What happens if the same contractId exists but for different municipalities? This method will then remove contracts for multiple municipalities, is that desired? 

The same goes for findByContractIdAndId and deleteByContractIdAndId (as there is a chance that the combo will exist for multiple municipalities, at least in theory).

Reply:
The contractId is "unique", i.e. it is generated and can only exist for one or more versions of the same contract.
No chance that difference municipalities may have the same contractId.